airflow/providers/databricks/hooks/databricks_base.py

```diff
@@ -502,6 +598,13 @@ async def _a_get_token(self, raise_error: bool = False) -> str | None:
             self.log.debug("Using AAD Token for managed identity.")
             await self._a_check_azure_metadata_service()
             return await self._a_get_aad_token(DEFAULT_DATABRICKS_SCOPE)
+        elif self.databricks_conn.extra_dejson.get(DEFAULT_AZURE_CREDENTIAL_SETTING_KEY, False):
+            if not self._running_in_kubernetes():
+                raise AirflowException(
+                    "Workload identity authentication is only supporting when running in an Kubernetes cluster"
+                )
+            self.log.debug("Using Azure Workload Identity authentication.")
```

Comment:
I'd change this message to indicate that you're using the AzureDefaultCredential instead of referencing workload identity. Workload identity is only one of the (many) mechanisms that the AzureDefaultCredential can use for authentication and we have no guarantee that it will actually use workload identity.